docs/docs/getting-started/usage/opencode.md

```diff
@@ -0,0 +1,205 @@
+
+# Global with builtins
+cupcake init --global --harness opencode --builtins system_protection,sensitive_data_protection
+```
+
+Available builtins include:
+
+- `git_pre_check` — Run checks before git operations
+- `git_block_no_verify` — Prevent `--no-verify` flag usage
+- `system_protection` — Protect system directories
+- `sensitive_data_protection` — Block access to sensitive files
+
+See the [Built-in Configuration Reference](../../reference/builtin-config/) for complete details.
+
+## Plugin Configuration
+
+Create `.cupcake/opencode.json` to customize behavior:
+
+```json
+{
+  "enabled": true,
+  "cupcakePath": "cupcake",
+  "harness": "opencode",
+  "logLevel": "info",
+  "timeoutMs": 5000,
+  "failMode": "closed",
+  "cacheDecisions": false
+}
+```
+
+| Option           | Default     | Description                                             |
+|------------------|-------------|---------------------------------------------------------|
+| `enabled`        | `true`      | Enable/disable the plugin                               |
+| `cupcakePath`    | `"cupcake"` | Path to cupcake binary                                  |
+| `logLevel`       | `"info"`    | Log level: debug, info, warn, error                     |
+| `timeoutMs`      | `5000`      | Max policy evaluation time (ms)                         |
+| `failMode`       | `"closed"`  | `"open"` (allow on error) or `"closed"` (deny on error) |
+| `cacheDecisions` | `false`     | Cache decisions (experimental)                          |
+
+## Tool Name Mapping
+
+OpenCode uses lowercase tool names. Cupcake normalizes them automatically:
+
+| OpenCode  | Cupcake Policy |
+|-----------|----------------|
+| `bash`    | `Bash`         |
+| `edit`    | `Edit`         |
+| `write`   | `Write`        |
+| `read`    | `Read`         |
+| `grep`    | `Grep`         |
+| `glob`    | `Glob`         |
+
+## Verify Installation
+
+### 1. Create a test event
+
+Save this to `test-event.json`:
+
+```json
+{
+  "hook_event_name": "PreToolUse",
+  "session_id": "test",
+  "cwd": "/tmp",
+  "tool": "bash",
+  "args": {
+    "command": "echo 'Hello from Cupcake!'"
+  }
+}
+```
+
+### 2. Evaluate the event
+
+```bash
+cupcake eval --harness opencode --policy-dir .cupcake/policies < test-event.json
+```
+
+### 3. Expected output
+
+```json
+{
+  "decision": "allow"
+}
+```
+
+## Key Differences from Other Harnesses
+
+| Aspect            | Claude Code / Cursor            | OpenCode                       |
+|-------------------|--------------------------------|--------------------------------|
+| Integration       | External hooks (stdin/stdout)   | In-process TypeScript plugin   |
+| Blocking          | Return JSON `{continue: false}` | Throw Error                    |
+| Ask Support       | Native                          | Converted to deny with message |
+| Context Injection | `additionalContext` field       | Limited (future enhancement)   |
```

Comment:
If this is important we should move this information to the "how it works" section at the top?

```suggestion
```